core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -766,7 +775,8 @@ class ReplicaManager(val config: KafkaConfig,
             }
             topicPartition -> LogAppendResult(
               LogAppendInfo.UNKNOWN_LOG_APPEND_INFO,
-              Some(finalException)
+              Some(finalException),
+              hasCustomErrorMessage = true
```

Comment:
Maybe this is ok -- but I think we are still returning custom error messages for non-custom messages in the final case of the match here. Error.exception I think uses the default message.

Reply:
I sort of want to refactor this code so maybe we can do that in a followup.

Reply:
hmm, I see what you are saying. I will fix it.